### ISSUE TITLE
[3.3.5] Problem with spells based on percentage of DoT or HoT

### ISSUE THREAD

**Issue by Keader** [TrinityCore/TrinityCore#14958]
Spell 1 : Swiftmend [Resto-Druid]
http://wotlk.openwow.com/spell=18562

Description: Consumes a Rejuvenation or Regrowth effect on a friendly target to instantly heal them an amount equal to 12 sec. of Rejuvenation or 18 sec. of Regrowth.

Problem: The swiftmend Cure value is below promised in the spell.

Test:

Using Rejuvenation rank 15 without the HoT spell power of the healing value is 422.

![reju1](https://cloud.githubusercontent.com/assets/7120913/8385365/266dffaa-1c1e-11e5-9462-1e3398bfa999.jpg)

At full duration (15 seconds) the spell should heal 5x 422.

![reju2](https://cloud.githubusercontent.com/assets/7120913/8385425/7b341934-1c1e-11e5-9e8b-c7a3215ad6aa.jpg)

Based on this, the swiftmend should convert Rejuvenation cure would in 12 seconds in a instant cure. Doing the math would be 5x422 = 2110. However the talent says it's 12 seconds of rejuvenation and not the total time. So one cure of rejuvenation is "lost". Then 2110-422 = 1688. However the result is as follows:

![swift1](https://cloud.githubusercontent.com/assets/7120913/8385555/61c94496-1c1f-11e5-9c3b-1f2394a5504a.jpg)

According to the Addon DrDamage healing should be 1690 (it takes into account the rejuvenation cure to 422.5)

![swift2](https://cloud.githubusercontent.com/assets/7120913/8385574/9075271a-1c1f-11e5-857c-3084eafcf763.jpg)

4 * 422 = 1688 != 1353
The build used pros tests is as follows:

![build druid](https://cloud.githubusercontent.com/assets/7120913/8385624/f02a0bf8-1c1f-11e5-965f-56729d140fa9.jpg)

Spell 2: Chimera Shot [MARKS-HUNTER]
http://wotlk.openwow.com/spell=53209

Description: You deal 125% weapon damage, refreshing the current Sting on your target and triggering an effect: Serpent Sting - Instantly deals 40% of the damage done by your Serpent Sting. Viper Sting - Instantly restores mana to you equal to 60% of the total amount drained by your Viper Sting. Scorpid Sting - Attempts to Disarm the target for 10 sec. This effect cannot occur more than once per 1 minute.

Problem: Damage percentage applied different than promised in the spell

Tests:

Using the serpenting shot rank 12 this is the damage:
![serpent](https://cloud.githubusercontent.com/assets/7120913/8385729/d5d5b8a0-1c20-11e5-8b7c-297d206b26d7.jpg)

Doing the calculations, 5x331 = 1655. 40% of damage 662. However the spell is presented by the following:

![chimera](https://cloud.githubusercontent.com/assets/7120913/8385789/3c5036d2-1c21-11e5-908b-6f15fe1d5833.jpg)

The Damage should be 663 however the value was 510.

Build used in the tests:
![build hunter](https://cloud.githubusercontent.com/assets/7120913/8385843/c1c9b2b6-1c21-11e5-9631-5b98490aa074.jpg)

Tests on rev accab5451d3e62739e4aa376d25bc6623f0d3c52
It maybe related to:  #14869 and #14918

**Comment by Keader** [comment]
@robinsch It could give a look at these spells later?

**Comment by robinsch** [comment]
Untested.

```diff
diff --git a/src/server/game/Spells/SpellEffects.cpp b/src/server/game/Spells/SpellEffects.cpp
index 3829e12..1f6a34b 100644
--- a/src/server/game/Spells/SpellEffects.cpp
+++ b/src/server/game/Spells/SpellEffects.cpp
@@ -1413,7 +1413,12 @@ void Spell::EffectHeal(SpellEffIndex /*effIndex*/)
 
             int32 tickheal = targetAura->GetAmount();
             if (Unit* auraCaster = targetAura->GetCaster())
+            {
                 tickheal = auraCaster->SpellHealingBonusDone(unitTarget, targetAura->GetSpellInfo(), tickheal, DOT);
+
+                if (auraCaster->GetTypeId() == TYPEID_PLAYER)
+                    auraCaster->ToPlayer()->ApplySpellMod(targetAura->GetSpellInfo()->Id, SPELLMOD_DOT, tickheal);
+            }
             //int32 tickheal = targetAura->GetSpellInfo()->EffectBasePoints[idx] + 1;
             //It is said that talent bonus should not be included
 ```

**Comment by Keader** [comment]
Swiftmend fix works \o/

**Comment by mik1893** [comment]
Is corruption still working properly without this?

**Comment by mik1893** [comment]
closed by #14987
